### PR TITLE
Use DeploymentId type from toolshed repo

### PR DESCRIPTION
**Pull request by Jannis**
This drops the custom `SubgraphDeploymentID` type in favor of the `DeploymentId` type from the https://github.com/edgeandnode/toolshed repository, which we're working on making sort of a Graph standard utility library.

Saves a bunch of code and is one more isolated piece of #54.

### CODE REVIEW

**Review by hopeyen — APPROVED** [review]
lgtm, just need to pass the formatting check

**Review by aasseman — APPROVED** [review]
Looks great, though I didn't really look into what's going on with `service/src/common/indexer_management/mod.rs`.

**Comment by Jannis** [comment]
> Looks great, though I didn't really look into what's going on with `service/src/common/indexer_management/mod.rs`.

@aasseman There's two main changes there:

1. There were a few issues between the deployments filter for cost models and what was returned (and merged with the global cost model).
2. I tried to strongly type the `deployment` field of the `CostModel` type and still make deserialization from the database work but gave up after two days (among other things, the special `"global"` deployment ID in the cost model table was annoying. So now there's three types:
  - `DbCostModel` is an internal type and is what is returned by the db queries, with deployment IDs being `String` (`"0x..."`).
  - `CostModel` is the public type returned by the `cost_models()` and `cost_model()` functions; this never includes `"global"` as the deployment, so `DeploymentId` worked well here.
  - `GqlCostModel` is a helper type to serialize deployments in cost models to IPFS hashes.

**Comment by hopeyen** [comment]
> There were a few issues between the deployments filter for cost models and what was returned (and merged with the global cost model).

Could you describe what the issues were for the previous cost model responses? I read through the test cases and they seemed to match with my understanding of how it worked

> So now there's three types

I can't seem to find `GqlCostModel` here or in toolshed, could you help me locate it 🤔 ?

**Comment by Jannis** [comment]
> > There were a few issues between the deployments filter for cost models and what was returned (and merged with the global cost model).
> 
> Could you describe what the issues were for the previous cost model responses? I read through the test cases and they seemed to match with my understanding of how it worked

One was that if you provided no deployments filter for `cost_model()`, it would return all existing cost models (including the `"global"` one, I think) without merging in the global model (https://github.com/graphops/indexer-service-rs/pull/63/files#diff-21ab5cb781d51883665196101d048fe83b746acdd69253df92bdedcfcf6cbba9L85).

It may be that that was the only one related to the database. Another was that the error handling around invalid deployment IDs passed in via GraphQL was a bit sketchy, and the conversion between IPFS hashes (required for the GraphQL API) and hex strings (required for the database) wasn't very robust. I tried to make those parts more obvious and robust.

> > So now there's three types
> 
> I can't seem to find `GqlCostModel` here or in toolshed, could you help me locate it 🤔 ?

@hopeyen Ooops, it's called `GraphQlCostModel`: https://github.com/graphops/indexer-service-rs/pull/63/files#diff-30f1dd4cff74e5decaff51bc44c6cea8ddc6f74813106dbe1c0fe609d5ca6b6bR19

**Comment by aasseman** [comment]
> > Looks great, though I didn't really look into what's going on with `service/src/common/indexer_management/mod.rs`.
> 
> @aasseman There's two main changes there:
> 
> 1. There were a few issues between the deployments filter for cost models and what was returned (and merged with the global cost model).
> 2. I tried to strongly type the `deployment` field of the `CostModel` type and still make deserialization from the database work but gave up after two days (among other things, the special `"global"` deployment ID in the cost model table was annoying. So now there's three types:
> 
> * `DbCostModel` is an internal type and is what is returned by the db queries, with deployment IDs being `String` (`"0x..."`).
> * `CostModel` is the public type returned by the `cost_models()` and `cost_model()` functions; this never includes `"global"` as the deployment, so `DeploymentId` worked well here.
> * `GqlCostModel` is a helper type to serialize deployments in cost models to IPFS hashes.

Thanks :pray: !

**Review by hopeyen — APPROVED** [review]
thanks for the explanation. looks good!

**Review by aasseman — APPROVED** [review]

File: service/src/common/indexer_management/mod.rs [hunk 1]
```diff
@@ -179,12 +219,29 @@ mod test {
...
+                VALUES ($1, $2);
+                "#,
+                format!("{deployment:#x}"),
+                model.model,
+            )
+            .execute(pool)
+            .await
+            .expect("Create test instance in db");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Jannis) [reply service/src/common/indexer_management/mod.rs:236]: I'll submit a follow-up PR to simply this a little!

**Pull request merged**